test/testexprengine.cpp

```diff
@@ -660,6 +661,19 @@ class TestExprEngine : public TestFixture {
                       expr(code, "=="));
     }
 
+    void forAlwaysFalse1() {
+        const char code[] = "int f() {\n"
+                            "  int a = 19;\n"
+                            "  for (int i = 0; i < 0; i++)\n"
+                            "    a += 8;\n"
+                            "  for (int i = 0; i < 1; i++)\n"
+                            "    a += 23;\n"
+                            "  return a == 42;\n"
+                            "}";
+        ASSERT_EQUALS("(= 42 42)\n"
+                      "z3::sat\n",
+                      expr(code, "=="));
+    }
```

Comment:
I am not sure how well you handle when the loop counter is decreased. Can you add a test case for that also. For instance;

    for (int i = 100; i >= 0; i--)

Reply:
Currently we handle only increasing `for` loops [here](https://github.com/danmar/cppcheck/blob/1e300a6ef4857012c5a872e02d4608ef96460248/lib/exprengine.cpp#L2663). This is due to the implementation of [`extractForLoopValues`](https://github.com/danmar/cppcheck/blob/d59164e9ef360fa487a6d723e72c26f76370955f/lib/astutils.cpp#L533).

Reply:
But in general, it's not a bad idea to add a test to avoid regressions in the future